Yaohong create Quick Setup Modal for new team members

YHX6 opened the pull request:
# Description
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/113729159/537d5a04-1d56-4dcd-bb67-8cbe4ff0b6bc)
This is a continuation of [#PR 1673](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1673)  and [#PR  644](https://github.com/OneCommunityGlobal/HGNRest/pull/644)

## Related PRS (if any):
To test this backend PR you need to checkout the [#2092 frontend PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2092).
…

## Main changes explained:
 - Created title schema
 - Created controller for interacting with database
 - Created route from title
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin or ownder user
5. go to dashboard→ user management→ click a user → user profile page
6. Task 1: Click a QSC to open a modal to enter information and then save changes
7. Task 2: Click another QSC to overwrite user profile information
8. Task 3: Click "Add a new title" button to create a title/QSC (make sure your team code exists); Click a QSC to delete it

## Screenshots or videos of changes:
Task 1: Click a QSC to open a modal to enter information and then save changes

https://github.com/OneCommunityGlobal/HGNRest/assets/113729159/2ec65fcf-1010-40a8-af39-42645cde53c5


Task 2: Click another QSC to overwrite user profile information


https://github.com/OneCommunityGlobal/HGNRest/assets/113729159/6ddc5093-7d51-4ad0-bb1d-1931b55d8fda



Task 3:  Click "Add a new title" button to create a title/QSC (make sure your team code exists); Click a QSC to delete it


https://github.com/OneCommunityGlobal/HGNRest/assets/113729159/cf95d36b-6d31-4f2a-9f72-99d02986604c

abdel-lall commented on src/controllers/titleController.js:
```diff
@@ -0,0 +1,51 @@
+      title.teamAssiged = req.body.teamAssiged;
+      // get the shortname
+      const shortnames = title.titleName.split(' ');
+      const shortname = (shortnames[0][0] + shortnames[1][0]).toUpperCase();
```
Hi @YHX6, Just want to bring to your attention to line of code. This will not throw an error if the title is only one word.
![Screenshot 2024-03-30 153715](https://github.com/OneCommunityGlobal/HGNRest/assets/49083865/4ddb4e4b-0371-4dd0-868d-d6f9c28d3053)

YHX6 replied:
Thanks! I'll fix that

j-corp-25 reviewed (commented):
Hi @YHX6,

I tested the feature and got some of the expected results,

Here are some of my notes

- Whenever Im creating a new title it doesnt always work, I have no way of knowing if its a missing field that im missing or filling the field incorrectly.
- I also was getting multiple key errors in the console. In the video below it shows me testing what was shown in the first video.


https://github.com/OneCommunityGlobal/HGNRest/assets/95154804/8b6107f8-f685-4b54-802b-359805019bb9

YHX6 commented:
The above issues of not showing the page occur because some of the titles created have problems. I have deleted all those titles and added validations in both the frontend and backend files. It works on my laptop right now

YHX6 commented:
As for the issue of unexpected keys, I am not sure where that comes from. However, the error seemed to exist before I worked on this PR. Maybe it is originated from other parts of code

Parth-tech reviewed (commented):
Dropped a review on the [frontend repo](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2092#pullrequestreview-1995534613)

SushmithaPrathap reviewed (approved):
Hey, I have tried and tested this PR, followed the steps mentioned.
Here is link to my frontend review:[ Frontend Link](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2092#pullrequestreview-1998769956)

Tiantian-C reviewed (commented):
It works fine!

https://github.com/OneCommunityGlobal/HGNRest/assets/122494725/ef7ff58e-85f8-4e31-bedd-816e3ca694ed

p-suram reviewed (approved):
Hey, Good Work. Changes look good. Please find my frontend PR Review [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2092#pullrequestreview-1999402066)

aaryaneil reviewed (approved):
Left a review on the [frontend](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2092#pullrequestreview-1999430945)

jennZhang93 reviewed (approved):
left a frontend review: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2092#pullrequestreview-2023868876

cvtqx reviewed (approved):
Works well. Left a review in the [FE PR.](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2092#pullrequestreview-2039554457)
